### PR TITLE
CEDS-5123 Create Feature Flag for improved error handling

### PR DESCRIPTION
**Pull request by glidester**

### CODE REVIEW

File: app/controllers/ErrorsReportedController.scala [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+    val messages = messagesApi.preferred(request).messages
+    Future.successful(Ok(rejectedNotificationPage()(request, messages)))
+  }
+
+  def displayPageOnUnacceptedAmendment(actionId: String, draftDeclarationId: Option[String] = None): Action[AnyContent] =
+    (authenticate andThen verifyEmail).async { implicit request =>
+      val messages = messagesApi.preferred(request).messages
+      Future.successful(Ok(rejectedNotificationPage()(request, messages)))
+    }
+}
```
Comment (udinobi) [inline app/controllers/ErrorsReportedController.scala:47]: Why do you need to define `messages`?
This controller could simply be reduced to...
```
  val displayPage: Action[AnyContent] = (authenticate andThen verifyEmail).async { implicit request =>
    Future.successful(Ok(rejectedNotificationPage()))
  }

  val displayPageOnUnacceptedAmendment: Action[AnyContent] = (authenticate andThen verifyEmail).async { implicit request =>
    Future.successful(Ok(rejectedNotificationPage()))
  }
```

File: app/controllers/RejectedNotificationsController.scala [hunk 2]
```diff
@@ -37,49 +38,61 @@ class RejectedNotificationsController @Inject() (
...
-          val mrn = notifications.headOption.map(_.mrn).getOrElse(messages("rejected.notification.mrn.missing"))
-          Ok(rejectedNotificationPage(None, declaration, mrn, None, getRejectedNotificationErrors(notifications)))
-        }
+    if (newErrorReportConfig.isNewErrorReportEnabled)
+      errorsReportedController.displayPage(id).apply(request)
+    else {
+      customsDeclareExportsConnector.findDeclaration(id).flatMap {
+        case Some(declaration) =>
+          customsDeclareExportsConnector.findNotifications(id).map { notifications =>
+            val messages = messagesApi.preferred(request).messages
```
Comment (udinobi) [inline app/controllers/RejectedNotificationsController.scala:54]: ???

File: app/controllers/RejectedNotificationsController.scala [hunk 3]
```diff
@@ -37,49 +38,61 @@ class RejectedNotificationsController @Inject() (
...
   def displayPage(id: String): Action[AnyContent] = (authenticate andThen verifyEmail).async { implicit request =>
-    customsDeclareExportsConnector.findDeclaration(id).flatMap {
-      case Some(declaration) =>
-        customsDeclareExportsConnector.findNotifications(id).map { notifications =>
-          val messages = messagesApi.preferred(request).messages
-          val mrn = notifications.headOption.map(_.mrn).getOrElse(messages("rejected.notification.mrn.missing"))
-          Ok(rejectedNotificationPage(None, declaration, mrn, None, getRejectedNotificationErrors(notifications)))
-        }
+    if (newErrorReportConfig.isNewErrorReportEnabled)
+      errorsReportedController.displayPage(id).apply(request)
```
Comment (udinobi) [inline app/controllers/RejectedNotificationsController.scala:49]: The `id` parameter is not used in `displayPage`

File: app/controllers/RejectedNotificationsController.scala [hunk 4]
```diff
@@ -37,49 +38,61 @@ class RejectedNotificationsController @Inject() (
...
-            case Some(decId) =>
-              val declarationId = draftDeclarationId.fold(decId)(identity)
-              customsDeclareExportsConnector.findDeclaration(declarationId).flatMap {
-                case Some(declaration) =>
-                  customsDeclareExportsConnector.findLatestNotification(actionId).map {
-                    case Some(notification) =>
-                      Ok(rejectedNotificationPage(getValue(submissionUuid), declaration, notification.mrn, Some(declarationId), notification.errors))
-                        .addingToSession(submissionActionId -> actionId)
+      if (newErrorReportConfig.isNewErrorReportEnabled)
+        errorsReportedController.displayPageOnUnacceptedAmendment(actionId, draftDeclarationId).apply(request)
```
Comment (udinobi) [inline app/controllers/RejectedNotificationsController.scala:67]: `actionId` and `draftDeclarationId` are not used in `displayPageOnUnacceptedAmendment`

File: app/controllers/ErrorsReportedController.scala (hunk 1 above)
Comment (glidester) [reply app/controllers/ErrorsReportedController.scala:45]: Removed :+1:

**Review by udinobi — APPROVED** [review]

**Pull request merged**